File: java/src/main/java/gov/doc/ntia/sigmf/ext/global/sensor/Preselector.java
package gov.doc.ntia.sigmf.ext.global.sensor;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import gov.doc.ntia.sigmf.ext.global.core.HardwareSpec;
import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Preselector implements Serializable {

  private static final long serialVersionUID = 1L;

  @JsonProperty(value = "preselector_spec", required = false)
  protected HardwareSpec preselectorSpec;

  @JsonProperty(value = "cal_sources", required = false)
  protected CalSource[] calSources;

  @JsonProperty(value = "filters", required = false)
  protected Filter[] filters;

  @JsonProperty(value = "amplifiers", required = false)
  protected Amplifier[] amplifiers;

  @JsonProperty(value = "rf_paths", required = false)
  protected RfPath[] rfPaths;

  protected Map<String, Object> otherFields = new HashMap<>();

  public HardwareSpec getPreselectorSpec() {
    return preselectorSpec;
  }

  public void setPreselectorSpec(HardwareSpec preselectorSpec) {
    this.preselectorSpec = preselectorSpec;
  }

  public CalSource[] getCalSources() {
    return calSources;
  }

  public void setCalSources(CalSource[] calSources) {
    this.calSources = calSources;
  }

  public Filter[] getFilters() {
    return filters;
  }

  public void setFilters(Filter[] filters) {
    this.filters = filters;
  }

  public Amplifier[] getAmplifiers() {
    return amplifiers;
  }

  public void setAmplifiers(Amplifier[] amplifiers) {
    this.amplifiers = amplifiers;
  }

  public RfPath[] getRfPaths() {
    return rfPaths;
  }

  public void setRfPaths(RfPath[] rfPaths) {
    this.rfPaths = rfPaths;
  }

  @JsonAnyGetter
  public Map<String, Object> getOtherFields() {
    return otherFields;
  }

  public void setOtherFields(Map<String, Object> otherFields) {
    this.otherFields = otherFields;
  }

  @JsonAnySetter
  public void add(String key, Object value) {
    otherFields.put(key, value);
  }
}

File: Csharp/sigmf/sigmf/ext/global/sensor/Preselector.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable()]
    public class Preselector
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("preselector_spec")]
        protected HardwareSpec preselectorSpec;

        [JsonProperty("cal_source_spec")]
        protected HardwareSpec calSourceSpec;

        [JsonProperty("lna_spec")]
        protected HardwareSpec lnaSpec;

        [JsonProperty("filter_spec")]
        protected HardwareSpec[] filterSpec;

        [JsonProperty("rf_paths")]
        protected RFPath[] rfPaths;

        protected Dictionary<string, Object> otherfields = new Dictionary<string, Object>();

        public HardwareSpec PreselectorSpec { get => preselectorSpec; set => preselectorSpec = value; }
        public HardwareSpec CalSourceSpec { get => calSourceSpec; set => calSourceSpec = value; }
        public HardwareSpec LnaSpec { get => lnaSpec; set => lnaSpec = value; }
        public HardwareSpec[] FilterSpec { get => filterSpec; set => filterSpec = value; }
        public RFPath[] RfPaths { get => rfPaths; set => rfPaths = value; }

        public Dictionary<string, object> getOtherFields()
        {
            return otherfields;
        }
        public void add(string key, object value)
        {
            otherfields.Add(key, value);
        }
    }
}
